content/fast/adaptive-serving-based-on-network-quality/index.md

```diff
@@ -0,0 +1,103 @@
+---
+page_type: guide
+title: Adaptive serving based on network quality
+author: mihajlija
+description: |
+  Using Network Information API to adapt which assets you're serving to users based on the quality of their connection.
+web_lighthouse: N/A
+web_updated_on: 2019-04-01
+web_published_on: 2019-04-01
+wf_blink_components: N/A
+---
+
+# Adaptive serving based on network quality
+
+Loading a website can be a very different experience depending on the network conditions. Everything is usually smooth when you are on a fast network, but when you are on the go with a humble data plan and spotty connection, or stuck with a laptop on a sluggish coffee shop wifi, it's a different story.
+
+One way to deal with this is by adapting which assets you're serving to users based on the quality of their connection. This is now [possible](https://caniuse.com/#feat=netinfo) with the [Network Information API](https://developer.mozilla.org/en-US/docs/Web/API/Network_Information_API) which enables web applications to access information about the user's network.
+
+## Usage
+
+There are many ways you can use this network information to improve the user experience:
+
```

Comment:
IMPORTANT: This list is interesting to me-- I want to learn about each of them. Is it possible for this guide to focus more on using the network information API to improve experience, rather than covering the what? Maybe even each codelab associated with the doc covers one of these list items?

Reply:
The thing with this list is that it's highly hypothetical at this point, there aren't (m)any real-world examples I can reference. I could expand on it by writing a couple of sentences about each of these items, but a list feels much more in line with web.dev writing guidelines?

As for the codelabs, some list items, such as warning users about data charges on cellular, are good candidates. Others, such as deferring uploads and downloads might be too complicated for the codelab/under 1000 words format. I'll discuss this with Addy in our weekly call 👍

Reply:
Hey @Meggin! I think Milica and I would love to see a deeper-dive on NetInfo usecases and code-labs. We'd like to circle back on this later in the year. Right now NetInfo usage is still quite low (we only found one big site using it on HTTP Archive) and I'd love to avoid us investing too deeply just yet on the topic when we have topic migrations we need to get to (like third-party resource loading) :) 

Does that make sense? Maybe we can file an issue to file-up. There's a lot of nuance here like some of these use-cases are quite involved to show a demo of and we'd want to noodle on the best way to present this before forging ahead with codelab content.